public/js/redux/chartsRedux.js

```diff
@@ -28,9 +30,15 @@ const chartsRedux = State({
         }
     },
 
-    updateComposerVsIncopy(state, composerVsInCopyData) {
+    updateComposerVsIncopy(state, { chartData, startDate, endDate }) {
+        const { composerResponse, inCopyResponse } = chartData;
+        const range = endDate.diff(startDate, 'days');
+        const composerData = composerResponse.data.length < range ?  fillMissingDates(startDate, endDate, composerResponse.data).sort(compareDates) : composerResponse.data;
```

Comment:
depends, is the data correct with 0 values or it is it a presentation concern? If the latter, then adding rendering logic to the API response could be harmful for other clients that use the API.